packages/grid_client/src/clients/kyc/client.ts

```diff
@@ -0,0 +1,131 @@
+
+/**
+ * The KYC class provides methods to interact with a TFGid KYC (Know Your Customer) service.
+ * It allows fetching verification data, status, and token by preparing necessary headers
+ * and sending requests to the specified API domain.
+ *
+ * @class KYC
+ * @example
+ * ```typescript
+ * const kyc = new KYC("https://api.example.com", "TFChain-address", KeypairType.sr25519, "mnemonic");
+ * const data = await kyc.data();
+ * const status = await kyc.status();
+ * const token = await kyc.token();
+ * ```
+ * @param {string} apiDomain - The API domain for the KYC service.
+ * @param {string} address - The TFChain address.
+ * @param {KeypairType} [keypairType=KeypairType.sr25519] - The type of keypair to use.
+ * @param {string} mnemonic - The mnemonic for generating the keypair.
+ * @method data - Fetches the verification data from the KYC service.
+ * @method status - Fetches the verification status from the KYC service.
+ * @method token - Fetches the token from the KYC service.
+ */
+export default class KYC {
+  private keyr: Keyring;
+  /**
+   * Creates an instance of KYC.
+   * @param apiDomain - The API domain for the TFGrid KYC service.
+   * @param address - The TFChain user address.
+   * @param keypairType - The type of keypair to use (default is sr25519).
+   * @param mnemonic - The mnemonic for generating the keypair.
+   */
+  constructor(
+    public apiDomain: string,
+    public address: string,
+    public keypairType: KeypairType = KeypairType.sr25519,
+    private mnemonic: string,
+  ) {
+    this.keyr = new Keyring({ type: keypairType });
+  }
+
+  /**
+   * Retrieves a key pair from the mnemonic.
+   *
+   * @returns {Promise<KeyringPair>} A promise that resolves to a KeyringPair object.
+   * @private
+   */
+  private async getKey(): Promise<KeyringPair> {
+    await waitReady();
+    return this.keyr.addFromUri(this.mnemonic);
+  }
+  /**
+   * Converts a string message to its hexadecimal representation.
+   * @param message - The message to convert.
+   * @returns The hexadecimal representation of the message.
+   */
+  private stringToHex(message: string): string {
+    return Buffer.from(message).toString("hex");
+  }
+  /**
+   * Converts a hexadecimal string to a byte array.
+   * @param hex - The hexadecimal string to convert.
+   * @returns {Uint8Array} The byte array representation of the hexadecimal string.
+   */
+  private bytesFromHex(hex: string): Uint8Array {
+    return new Uint8Array(Buffer.from(hex, "hex"));
+  }
+
+  /**
+   * Prepares the headers required for TFGrid KYC requests.
+   *
+   * This method generates a set of headers that include a timestamp-based challenge,
+   * a signed challenge using the user's key, and other necessary information.
+   *
+   * @returns {Promise<Record<string, string>>} A promise that resolves to an object containing the prepared headers.
+   *
+   */
+  private async prepareHeaders(): Promise<Record<string, string>> {
```

Comment:
why don't we use `KycHeaders` as the return type?

Reply:
the send method accepts `Record<string,string>` I'm using header type here to make it understandable only